src/Orleans.Core/Streams/QueueAdapters/IQueueAdapterFactory.cs

```diff
@@ -26,12 +24,5 @@ public interface IQueueAdapterFactory
         /// </summary>
         /// <returns></returns>
         IStreamQueueMapper GetStreamQueueMapper();
-
-        /// <summary>
-        /// Acquire delivery failure handler for a queue
-        /// </summary>
-        /// <param name="queueId"></param>
-        /// <returns></returns>
-        Task<IStreamFailureHandler> GetDeliveryFailureHandler(QueueId queueId);
```

Comment:
any chance this pr is extended and the whole IQueueAdapterFactory is replaced by DI?

Reply:
That's roughly the plan, long term.

The queue adapter was just supposed to be a thin abstraction layer over the underlying queuing system to allow the persistent stream infrastructure to run on a range of queuing solutions.  Over time and under delivery pressures it took on more features and complexity, in part because there was no uniform factory pattern in Orleans at the time.

Streams was one of the systems that prompted the adoption of DI, as it necessitated either the development of our own factory patterns or leveraging existing solutions.  DI was selected, and we've been (slowly) refactoring streams and other systems over time to leverage it.

There is a fair amount of technical debt in our streaming infrastructure, but the systems work sufficiently well to rarely become the priority.  We welcome help cleaning it up! :)